### PR TITLE
DAO Test Coverage

### PR DESCRIPTION
**Pull request by jorg3lopez**
# DAO Test Coverage (PostgresDao)

This PR adds additional test cases to cover the `fetchMetaData` in `PostgresDao` method.

## Issue
#672

### CODE REVIEW

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy [hunk 1]
```diff
@@ -137,4 +137,59 @@ class PostgresDaoTest extends Specification {
         then:
         thrown(SQLException)
     }
+
+    def "fetchMetadata happy path with hasValidData:true works"() {
```
Comment (jcrichlake) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:141]: Could we update the test names to be more business use case specific? Like "fetchMetadata succeeds when rows exist to return"?

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy [hunk 2]
```diff
@@ -137,4 +137,59 @@ class PostgresDaoTest extends Specification {
...
+        given:
+        def mockDriver = Mock(SqlDriverManager)
+        def mockPreparedStatement = Mock(PreparedStatement)
+        def mockResultSet = Mock(ResultSet)
+        def expected = null
+        Connection MockConn = Mock(Connection)
```
Comment (basiliskus) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:147]: Minor detail, but is there any reason `MockConn` name starts with uppercase? Same for the other test

**Review by basiliskus — APPROVED** [review]

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy (hunk 1 above)
Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy]: Yes, we can.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy [hunk 3]
```diff
@@ -137,4 +137,59 @@ class PostgresDaoTest extends Specification {
...
+
+        TestApplicationContext.register(SqlDriverManager, mockDriver)
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        def actual = PostgresDao.getInstance().fetchMetadata("mock_lookup")
+
+        then:
+        actual == expected
+    }
```
[the remark this answers is not in the document]
Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:194]: Done.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy (hunk 2 above)
Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy]: Good catch, done.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy (hunk 1 above)
Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy]: Done.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:194]: I like that, done.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy [hunk 4]
```diff
@@ -137,4 +137,59 @@ class PostgresDaoTest extends Specification {
...
+
+    def "fetchMetadata returns partnermetadata when rows exist"() {
+        given:
+        def mockDriver = Mock(SqlDriverManager)
+        def mockPreparedStatement = Mock(PreparedStatement)
```
Comment (rajeev-flexion) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:167]: We can put mocks setup in "def setup() {}" on the top.

Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy]: Yes, we can.

Comment (jorg3lopez) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy]: Done.

**Review by jcrichlake — APPROVED** [review]

**Pull request merged**